tools/test/examples/examples_lib.py

```diff
@@ -380,31 +382,60 @@ def compile_repos(config, toolchains, targets, profile, verbose, examples):
         successes = []
         compiled = True
         pass_status = True
+        if example.has_key('test') and example.has_key('baud_rate') and example.has_key('compare_log'):
+            test_example = True
+        else:
+            test_example = False
         if example['compile']:
             for repo_info in get_repo_list(example):
                 name = basename(repo_info['repo'])
                 os.chdir(name)
-
                 # Check that the target, toolchain and features combinations are valid and return a
                 # list of valid combinations to work through
                 for target, toolchain in target_cross_toolchain(valid_choices(example['targets'], targets),
                                                                 valid_choices(example['toolchains'], toolchains),
                                                                 example['features']):
-                    print("Compiling %s for %s, %s" % (name, target, toolchain))
+                    
                     build_command = ["mbed-cli", "compile", "-t", toolchain, "-m", target] + (['-v'] if verbose else [])
-
                     if profile:
                         build_command.append("--profile")
                         build_command.append(profile)
-
-                    proc = subprocess.Popen(build_command)
-
-                    proc.wait()
+                    
+                    print("Compiling [%s] for [%s] with toolchain [%s]\n\n>  %s" % (name, target, toolchain, " ".join(build_command)))
+                    
+                    proc = subprocess.Popen(build_command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+                    std_out, std_err = proc.communicate()
+                    print ("\n#### STDOUT ####\n%s\n#### STDERR ####\n%s\n#### End of STDOUT/STDERR ####\n" % (std_out,std_err))
+                    
+                    if test_example:
+                        log = example['compare_log'].pop(0)
```

Comment:
Any reason this needs to be a `pop` vs the following:
```suggestion
                        log = example['compare_log']
```

Reply:
Here `pop()` is used because the value of `compare_log` will be a list of logs files.
The reason for that is because some of the example repos contains more than one example, e.g. the `mbed-os-example-tls` and `mbed-os-example-ble`. So the each log file need to match with 
 the individual sub-example.  So we need to pop the file out of list regardless the compilation for each sub-example pass of fail

Reply:
May I request that this comment is added to the code please.